core-web/libs/utils/src/lib/dot-utils.ts

```diff
@@ -94,3 +94,24 @@ export function getImageAssetUrl(contentlet: DotCMSContentlet): string {
             return contentlet?.asset || '';
     }
 }
+
+/**
+ * This method is used to truncate a text with ellipsis.
+ * It ensures that the text is truncated at the nearest word boundary.
+ * @param text - The text to be truncated.
+ * @param limit - The maximum length of the truncated text.
+ * @returns The truncated text with ellipsis if it exceeds the limit, otherwise the original text.
+ */
+export function ellipsizeText(text: string, limit: number): string {
+    if (!text) {
+        return '';
+    }
+
+    if (text.length <= limit) {
+        return text;
+    }
+
+    const truncated = text.slice(0, limit);
+
+    return truncated.slice(0, truncated.lastIndexOf(' ')) + '...';
+}
```

Comment:
Take a look at this function, remember to validate if it is negative or not a number.

```javascript
const truncate = (text, limit = 25) => {
    const regexString = `^.{0,${limit}}`; // This will only match any char from the start of the string to the limit

    const regex = new RegExp(regexString, 'g'); // Create the regex

    return text.match(regex)[0]; // Get the match 
};
```

Reply:
Here are some test cases you can test:

```javascript
console.log(truncate('This is a test', 4)); // 'This'

console.log(truncate('This is a test', 5)); // 'This '

console.log(truncate('This is a test', 6)); // 'This i'

console.log(truncate('This is a test', 0)); // ''

console.log(truncate('This is a test', 30)); // 'This i a test'

```

Reply:
I believe that if there are spaces in the text, the truncation should occur at the previous word rather than splitting words. It looks better to me this way.